apollo-backend/src/test/java/io/logz/apollo/EnvironmentsStackTest.java

```diff
@@ -0,0 +1,87 @@
+    private static EnvironmentsStack environmentsStack;
+    private static int stackId;
+    private static final String NEW_STACK_NAME = "new-stack-name";
+    private static final boolean NEW_ENABLED = false;
+
+    @BeforeClass
+    public static void beforeClass() throws Exception {
+        standaloneApollo = StandaloneApollo.getOrCreateServer();
+        apolloTestClient = Common.signupAndLogin();
+
+        environmentsSize = (int) (Math.random() * 9 + 1);
+        for (int i = 0; i < environmentsSize; i++) {
+            environmentId = ModelsGenerator.createAndSubmitEnvironment(apolloTestClient).getId();
+            environments.add(environmentId);
+        }
+        environmentsStack = ModelsGenerator.createAndSubmitEnvironmentsStack(apolloTestClient, environments);
+        stackId = environmentsStack.getId();
+    }
+
+    @Test
+    public void testAddAndGetEnvironmentsStack() throws Exception {
+        apolloTestClient.removeEnvironmentFromStack(environmentId, stackId);
+        assertThat(apolloTestClient.getEnvironmentsStack(stackId).getEnvironments()).doesNotContain(environmentId);
+        apolloTestClient.addEnvironmentToStack(environmentId, stackId);
+        assertThat(apolloTestClient.getEnvironmentsStack(stackId).getEnvironments()).contains(environmentId);
+    }
+
+    @Test
+    public void testUpdateEnvironmentsStack() throws Exception {
+        List<Integer> newEnvironmentsList = new ArrayList<>();
+        for (int i = 0; i < environmentsSize; i++) {
+            newEnvironmentsList.add(ModelsGenerator.createAndSubmitEnvironment(apolloTestClient).getId());
+        }
+        environmentsStack.setEnvironments(newEnvironmentsList);
+        environmentsStack.setName(NEW_STACK_NAME);
+        environmentsStack.setEnabled(NEW_ENABLED);
+        apolloTestClient.updateEnvironmentsStack(environmentsStack);
+        EnvironmentsStack environmentsStackFromDb = apolloTestClient.getEnvironmentsStack(stackId);
+        this.assertThatEnvironmentsStacksEquals(environmentsStack, environmentsStackFromDb);
+    }
+
+    @Test
+    public void testGetAllEnvironmentsStacks() throws Exception {
+        List<Integer> environmentsStacksIds = new ArrayList<>();
+        int environmentsStacksAmount = (int) (Math.random() * 9 + 1);
+        for (int i = 0; i < environmentsStacksAmount; i++) {
+            environmentsStacksIds.add(ModelsGenerator.createAndSubmitEnvironmentsStack(apolloTestClient).getId());
+        }
+        List<EnvironmentsStack> allEnvironmentsStacks = apolloTestClient.getAllEnvironmentsStacks();
+        List<Integer> allEnvironmentsStacksIds = allEnvironmentsStacks.stream()
+                                                                      .map(environmentsStack -> environmentsStack.getId())
+                                                                      .collect(Collectors.toList());
+        environmentsStacksIds.stream().forEach(id -> {
```

Comment:
Why did you collect? you can continue the stream.
Or even better, isnt there a compare unordered list in assertThat?